lib/bugsnag/report.rb

```diff
@@ -49,14 +50,15 @@ def initialize(exception, passed_configuration, auto_notify=false)
       self.raw_exceptions = generate_raw_exceptions(exception)
       self.exceptions = generate_exception_list
 
-      self.api_key = configuration.api_key
-      self.app_type = configuration.app_type
-      self.app_version = configuration.app_version
+      self.api_key = configuration.api_key.dup
+      self.app_type = configuration.app_type.dup
+      self.app_version = configuration.app_version.dup
```

Comment:
If I apply the following diff, I can reproduce and inspect the failing test (by running `bundle exec rspec spec/integrations/rake_spec.rb` under Ruby 2.3.8):
```diff
--- a/spec/integrations/rake_spec.rb
+++ b/spec/integrations/rake_spec.rb
@@ -59,7 +59,7 @@ describe "Bugsnag Rake integration" do
       ENV['BUGSNAG_TEST_SERVER_PORT'] = server.config[:Port].to_s
       task_fixtures_path = File.join(File.dirname(__FILE__), '../fixtures', 'tasks')
       Dir.chdir(task_fixtures_path) do
-        system("bundle exec rake test:crash > /dev/null 2>&1")
+        system("bundle exec rake test:crash")
       end

       result = request()
```

The relevant resulting output is:
```
rake aborted!
TypeError: can't dup NilClass
/Users/toby/work/bugsnag-ruby/lib/bugsnag/report.rb:55:in `dup'
/Users/toby/work/bugsnag-ruby/lib/bugsnag/report.rb:55:in `initialize'
/Users/toby/work/bugsnag-ruby/lib/bugsnag.rb:70:in `new'
/Users/toby/work/bugsnag-ruby/lib/bugsnag.rb:70:in `notify'
/Users/toby/work/bugsnag-ruby/lib/bugsnag/integrations/rake.rb:21:in `rescue in execute'
/Users/toby/work/bugsnag-ruby/lib/bugsnag/integrations/rake.rb:31:in `execute'
/Users/toby/work/bugsnag-ruby/bundle/ruby/2.3.0/gems/rake-12.3.3/exe/rake:27:in `<top (required)>'
/Users/toby/.rbenv/versions/2.3.8/bin/bundle:22:in `load'
/Users/toby/.rbenv/versions/2.3.8/bin/bundle:22:in `<main>'
```

In Ruby 2.3 and lower, calling `dup` on `nil` fails. Since this pull request already has more than 20 files changed, I would suggest just using `dup` on your change (`configuration.runtime_versions.dup`) for now to avoid possible issues.

Reply:
Thanks @tobyhs!